src/monsters.cpp

```diff
@@ -431,31 +431,48 @@ bool Monsters::deserializeSpell(const pugi::xml_node& node, spellBlock_t& sb, co
 			combat->setParam(COMBAT_PARAM_TYPE, COMBAT_HEALING);
 			combat->setParam(COMBAT_PARAM_AGGRESSIVE, 0);
 		} else if (tmpName == "speed") {
-			int32_t speedChange = 0;
+			int32_t minSpeedChange = 0;
+			int32_t maxSpeedChange = 0;
 			int32_t duration = 10000;
 
 			if ((attr = node.attribute("duration"))) {
 				duration = pugi::cast<int32_t>(attr.value());
 			}
 
 			if ((attr = node.attribute("speedchange"))) {
-				speedChange = pugi::cast<int32_t>(attr.value());
-				if (speedChange < -1000) {
-					//cant be slower than 100%
-					speedChange = -1000;
+				minSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+				maxSpeedChange = minSpeedChange;
+			} else if ((attr = node.attribute("minspeedchange"))) {
+				minSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+				maxSpeedChange = minSpeedChange;
+
+				attr = node.attribute("maxspeedchange");
+
+				if (attr) {
+					maxSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+				}
+
+				if (!hasSameSign(minSpeedChange, maxSpeedChange)) {
```

Comment:
No need for this new function, `(minSpeedChange ^ maxSpeedChange) < 0` means signals differ.